code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -234,47 +316,53 @@ private String getActionPrefix(final boolean isInternalAction) {
+            return;
+        }
+        final HashMap<String, String> analyticsVars = processAnalyticsVars(event, data);
+        final HashMap<String, String> analyticsData = processAnalyticsData(event, data);
         sendAnalyticsHit(analyticsVars, analyticsData);
     }
 
     /**
      * This method converts the event's event data into analytics variables.
      *
      * @param event The Generic Track Request Content {@link Event}.
+     * @param data Track data for processing
      *
      * @return {@code Map<String, String>} containing the vars data
      */
-    private HashMap<String, String> processAnalyticsVars(final Event event) {
+    private HashMap<String, String> processAnalyticsVars(final Event event, final Map<String, Object> data) {
         final HashMap<String, String> processedVars = new HashMap<>();
-        final EventData eventData = event.getData();
-
-        if(eventData == null || eventData.isEmpty()) {
-            return processedVars;
-        }
-
-        // context: pe/pev2 values should always be present in track calls if there's action regardless of state.
+        // Context: pe/pev2 values should always be present in track calls if there's action regardless of state.
         // If state is present then pageName = state name else pageName = app id to prevent hit from being discarded.
-        final String actionName = eventData.optString(AnalyticsConstants.EventDataKeys.TRACK_ACTION, null);
+        final String actionName = (String) data.get(AnalyticsConstants.EventDataKeys.TRACK_ACTION);
         if(!StringUtils.isNullOrEmpty(actionName)) {
             processedVars.put(AnalyticsConstants.AnalyticsRequestKeys.IGNORE_PAGE_NAME, AnalyticsConstants.IGNORE_PAGE_NAME_VALUE);
-            boolean isInternal = eventData.optBoolean(AnalyticsConstants.EventDataKeys.TRACK_INTERNAL, false);
+            boolean isInternal = false;
+            if(data.get(AnalyticsConstants.EventDataKeys.TRACK_INTERNAL) != null) {
+                isInternal = (boolean) data.get(AnalyticsConstants.EventDataKeys.TRACK_INTERNAL);
+            }
             processedVars.put(AnalyticsConstants.AnalyticsRequestKeys.ACTION_NAME, getActionPrefix(isInternal) + actionName);
         }
-        // Todo :- We currently read application id from lifecycle
-        // processedVars.put(AnalyticsConstants.AnalyticsRequestKeys.PAGE_NAME, state->GetApplicationId());
-        final String stateName = eventData.optString(AnalyticsConstants.EventDataKeys.TRACK_STATE, null);
+
+        processedVars.put(AnalyticsConstants.AnalyticsRequestKeys.PAGE_NAME, getApplicationIdentifier());
+
+        final String stateName = (String) data.get(AnalyticsConstants.EventDataKeys.TRACK_STATE);
         if(!StringUtils.isNullOrEmpty(stateName)) {
             processedVars.put(AnalyticsConstants.AnalyticsRequestKeys.PAGE_NAME, stateName);
         }
```

Comment:
can we put `processedVars.put(AnalyticsConstants.AnalyticsRequestKeys.PAGE_NAME, getApplicationIdentifier());` in the else condition.